src/loader/cmi.ml

```diff
@@ -258,9 +335,7 @@ let prepare_type_parameters params manifest =
     | Some ty ->
         let vars = Ctype.free_variables ty in
           List.iter
-            (function {desc = Tvar (Some "_"); _} as ty ->
-              if List.memq ty vars then tvar_none ty
-                    | _ -> ())
+            (fun v -> if get_desc v = Tvar (Some "_") then (if List.memq ty vars then tvar_none ty))
```

Comment:
This is not equavalent. The `ty` before and after is not the same one (see `as ty` above)
```suggestion
            (fun ty -> match get_desc ty with
              | Tvar (Some "_") -> if List.memq ty vars then tvar_none ty
              | _ -> ())
```